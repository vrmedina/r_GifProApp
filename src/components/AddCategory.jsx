import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({onNewCategory}) => {
    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({ target }) => {
        setInputValue(target.value);
    };

    const onSubmit = (event) => {
        event.preventDefault();
        const newInputValue = inputValue.trim();
        if(newInputValue.length <= 1) return;
        setInputValue('');
        onNewCategory(newInputValue);
        // setCategories(categories => [inputValue, ...categories])
    };

    return (
        <form onSubmit={onSubmit} aria-label="form">
            <input 
                type="text"
                placeholder="Buscar Gifs"
                value={inputValue}
                onChange={onInputChange}
            />
            <button onClick={onSubmit}>Buscar</button>
        </form>
    );
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}

AddCategory.defaultProps = {
}