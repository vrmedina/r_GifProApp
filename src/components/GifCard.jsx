import PropTypes from 'prop-types';

const trimTitle = (string, limit = 30) => string.substring(0, limit)

export const GifCard = ({ title, url }) => {
    return (
        <div className='card'>
            <img src={url} alt={title}/>
            <p>{ trimTitle(title) }</p>
        </div>
    )
}

GifCard.propTypes = {
    title: PropTypes.string.isRequired,
    url:   PropTypes.string.isRequired
}

GifCard.defaultProps = {
}