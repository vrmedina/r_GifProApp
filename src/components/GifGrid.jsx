import PropTypes from 'prop-types';
import { GifCard } from './GifCard';

import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({category, limit}) => {

    const {gifs, isLoading} = useFetchGifs( category, limit );

    return (
        <>
            <h3>{category}</h3>

            { isLoading && <h2>Cargando tus Gifs...</h2> }

            <div className='card-grid'>
                {
                    gifs.map( gif => 
                        <GifCard 
                            key={gif.id} 
                            { ...gif }
                        />
                    )
                }
            </div>
        </>
    )
};

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
    limit: PropTypes.number.isRequired
}

GifGrid.defaultProps = {
}