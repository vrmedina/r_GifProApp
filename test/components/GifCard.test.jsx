import { render, screen } from '@testing-library/react';
import { GifCard } from '../../src/components/GifCard';

describe('Pruebas en <GifCard/>', () => {
    const title = 'Test Prueba';
    const url = 'https://www.test.test/';

    test('debe coincidir con el snapshot', () => {
        const {container} = render(<GifCard title={title} url={url}/>);
        expect(container).toMatchSnapshot();
    });

    test('debe mostrar la img con el alt y el url indicados', () => {
        render(<GifCard title={title} url={url}/>);
        const { alt, src } = screen.getByRole('img');
        expect( alt ).toBe(title);
        expect( src ).toBe(url);
    });

    test('debe mostrar el titulo en el componente', () => {
        render(<GifCard title={title} url={url}/>);
        expect( screen.getByText(title) ).toBeTruthy();  
    });
});
