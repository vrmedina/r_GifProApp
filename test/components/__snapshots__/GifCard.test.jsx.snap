// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifCard/> debe coincidir con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Test Prueba"
      src="https://www.test.test/"
    />
    <p>
      Test Prueba
    </p>
  </div>
</div>
`;
